samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,756 @@
+ */
+static int ota_wifi_connect(const char *ssid, const char *pwd, struct app_ctx *ctx)
+{
+	struct wifi_connect_req_params cnx_params = {
+		.channel = WIFI_CHANNEL_ANY,
+		.band = 0,
+		.security = ctx->security_type,
+		.psk_length = strlen(pwd),
+		.psk = (uint8_t *)pwd,
+		.ssid_length = strlen(ssid),
+		.ssid = (uint8_t *)ssid
+	};
+	struct net_if *iface = net_if_get_first_wifi();
+	int ret;
+
+	if (!ssid || !pwd) {
+		printf("Invalid connection parameters\n");
+		return -EINVAL;
+	}
+
+	printf("Connecting to network: %s\n", ssid);
+	ret = net_mgmt(NET_REQUEST_WIFI_CONNECT, iface, &cnx_params,
+		       sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		printf("Connection request failed with error: %d\n", ret);
+		return ret;
+	}
+	return 0;
+}
+
+/*
+ * @brief Wait for IP configuration to complete
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_wait_for_ip_config(struct app_ctx *ctx)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_NET_IPV4)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV4_EVENT_READY, false, K_SECONDS(30));
+	} else {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_SECONDS(30));
+	}
+	if (!ret && IS_ENABLED(CONFIG_NET_IPV6)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_NO_WAIT);
+	}
+	return (ret > 0) ? 0 : -ETIMEDOUT;
+}
+
+/*
+ * @brief Connect to OTA server
+ *
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_connect_to_server(struct app_ctx *ctx)
+{
+	struct http_parse_data http_parse_data_info = ctx->http_parse_data_info;
+	int peer_verify = TLS_PEER_VERIFY_NONE;
+	struct zsock_addrinfo *res;
+	int ret;
+
+	ret = zsock_getaddrinfo(http_parse_data_info.host, http_parse_data_info.port, NULL, &res);
+	if (ret != 0) {
+		printf("Address resolution failed: %s\n", zsock_gai_strerror(ret));
+		return ret;
+	}
+
+	/* Check if TLS is needed based on schema */
+	if (strcmp(http_parse_data_info.schema, "https") == 0) {
+		ctx->sock = zsock_socket(res->ai_family, res->ai_socktype, IPPROTO_TLS_1_2);
+		if (ctx->sock < 0) {
+			goto error;
+		}
+		ret = zsock_setsockopt(ctx->sock, SOL_TLS, TLS_PEER_VERIFY, &peer_verify,
+				       sizeof(peer_verify));
+		if (ret < 0) {
+			goto error;
+		}
+	} else {
+		/* Use regular TCP for HTTP */
+		ctx->sock = zsock_socket(res->ai_family, res->ai_socktype, IPPROTO_TCP);
+		if (ctx->sock < 0) {
+			goto error;
+		}
+	}
+
+	/* Connect to the server */
+	ret = zsock_connect(ctx->sock, res->ai_addr, res->ai_addrlen);
+	if (ret < 0) {
+		printf("Connection failed (%d): %s\n", -errno, strerror(errno));
+		ctx->sock = -1;
+		goto error;
+	}
+
+	printf("Connected to %s:%s\n", http_parse_data_info.host, http_parse_data_info.port);
+	zsock_freeaddrinfo(res);
+	return ctx->sock;
+
+error:
+	if (ctx->sock < 0) {
+		zsock_close(ctx->sock);
```

Comment:
hu? probably you mean `if (ctx->sock >= 0)`?

In fact, even this test is useless since zsock_close() on negative value will properly catch the error.